src/reps/promise.js

```diff
@@ -31,12 +31,7 @@ const PromiseRep = React.createClass({
 
   getTitle: function (object) {
     const title = object.class;
-    if (this.props.objectLink) {
-      return this.props.objectLink({
-        object: object
-      }, title);
-    }
-    return title;
+    return this.safeObjectLink({}, title);
```

Comment:
this looks like it can be refactored a bit:

perhaps: `safeObjectLink(this.props.objectLink, {}, title)`
or `this.props.safeObjectLink({}, title)`
 
to avoid some duplication